deltachat-ios/Controller/AccountSetup/Instand Onboarding/InstantOnboardingViewController.swift

```diff
@@ -1,78 +1,50 @@
-        return AvatarSelectionCell(image: dcContext.getSelfAvatarImage())
-    }()
-
-    private lazy var nameCell: TextFieldCell = {
-        let cell = TextFieldCell(description: String.localized("pref_your_name"), placeholder: String.localized("pref_your_name"))
-        cell.setText(text: dcContext.displayname)
-        cell.textFieldDelegate = self
-        cell.textField.returnKeyType = .default
-        return cell
-    }()
-
-    private lazy var sections: [SectionConfigs] = {
-        let nameSection = SectionConfigs(
-            headerTitle: nil,
-            footerTitle: String.localized("set_name_and_avatar_explain"),
-            cells: [nameCell, avatarSelectionCell]
-        )
-        return [nameSection]
-    }()
-
-    private lazy var acceptAndCreateButton: UIBarButtonItem = {
-        let button = UIBarButtonItem(
-            title: String.localized("instant_onboarding_create"),
-            style: .done,
-            target: self,
-            action: #selector(acceptAndCreateButtonPressed))
-        return button
-    }()
-
     init(dcAccounts: DcAccounts) {
         self.dcAccounts = dcAccounts
         self.dcContext = dcAccounts.getSelected()
-        super.init(style: .grouped)
-        hidesBottomBarWhenPushed = true
-    }
 
-    required init?(coder aDecoder: NSCoder) {
-        fatalError("init(coder:) has not been implemented")
-    }
+        super.init(nibName: nil, bundle: nil)
 
-    override func viewDidLoad() {
-        super.viewDidLoad()
-        navigationItem.rightBarButtonItem = acceptAndCreateButton
+        hidesBottomBarWhenPushed = true
         title = String.localized("pref_profile_info_headline")
-        avatarSelectionCell.onAvatarTapped = { [weak self] in
-            self?.onAvatarTapped()
-        }
-        tableView.rowHeight = UITableView.automaticDimension
     }
 
-    override func viewWillDisappear(_ animated: Bool) {
-        dcContext.displayname = nameCell.getText()
+    required init?(coder aDecoder: NSCoder) { fatalError("init(coder:) has not been implemented") }
+
+    override func loadView() {
+        super.loadView()
+
+        let contentView = InstantOnboardingView(avatarImage: dcContext.getSelfAvatarImage())
+        contentView.agreeButton.addTarget(self, action: #selector(InstantOnboardingViewController.acceptAndCreateButtonPressed), for: .touchUpInside)
+        contentView.imageButton.addTarget(self, action: #selector(InstantOnboardingViewController.onAvatarTapped), for: .touchUpInside)
+        contentView.privacyButton.addTarget(self, action: #selector(InstantOnboardingViewController.showPrivacy(_:)), for: .touchUpInside)
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(InstantOnboardingViewController.textDidChangeNotification(notification:)),
+            name: UITextField.textDidChangeNotification,
+            object: contentView.nameTextField
+        )
```

Comment:
don't we need to remove the observer again?

at many places, (also in code not caused by us :) ` `NotificationCenter.default.addObserver` is balanced by a`NotificationCenter.default.removeObserver()`